stm/stm.ml

```diff
@@ -2357,6 +2360,14 @@ let known_state ~doc ?(redefine_qed=false) ~cache id =
       try f x
       with e when CErrors.noncritical e -> () in
 
+  (* extract proof_ending in qed case, note that `Abort` and `Proof
+     term.` could also fail in this case, however that'd be a bug I do
+     believe as proof injection shouldn't happen here. *)
+  let extract_pe (x : aast) =
```

Comment:
We could remove this if we got rid of `Save` (see 3 commits including to 99d272fdc831e8371a5e5ffcd2b149fa2f9582a7)
If we want to keep Save we could also extract the idopt next to the vernac_keep_as when classifying.
Not sure if we want to do either so I didn't push to your branch.

Reply:
For sure all the commits there look look good. I guess you'll push the opaque fix as a PR?

Reply:
I am gonna take the first commit in this PR as anyways we need an overlay so it seems.